lib/screenplay_web/router.ex

```diff
@@ -1,156 +1,118 @@
   end
 
-  pipeline :ensure_screens_admin do
-    plug(ScreenplayWeb.EnsureScreensAdminGroup)
+  pipeline :ensure_outfront_admin, do: plug(EnsureRole, :emergency_admin)
+  pipeline :ensure_pa_message_admin, do: plug(EnsureRole, :pa_message_admin)
+  pipeline :ensure_screens_admin, do: plug(EnsureRole, :screens_admin)
+
+  # Load balancer health check, exempt from authentication, SSL redirects, etc.
+  scope "/_health", ScreenplayWeb do
+    get("/", HealthController, :index)
   end
 
-  # Load balancer health check
-  # Exempt from auth checks and SSL redirects
-  scope "/", ScreenplayWeb do
-    get("/_health", HealthController, :index)
+  scope "/auth", ScreenplayWeb do
+    pipe_through([:browser])
+
+    get("/:provider", AuthController, :request)
+    get("/:provider/callback", AuthController, :callback)
   end
 
-  scope "/", ScreenplayWeb.OutfrontTakeoverTool do
-    pipe_through([
-      :redirect_prod_http,
-      :browser,
-      :auth,
-      :ensure_auth,
-      :metadata,
-      :ensure_screenplay_emergency_admin_group
-    ])
+  scope "/unauthorized", ScreenplayWeb do
+    pipe_through([:browser, :authenticate])
 
-    get("/emergency-takeover", PageController, :index)
+    get("/", UnauthorizedController, :index)
   end
 
+  # Dashboard (Places/Alerts)
+
   scope "/", ScreenplayWeb do
-    pipe_through([:redirect_prod_http, :browser, :auth, :ensure_auth, :metadata])
+    pipe_through([:browser, :authenticate])
 
     get("/", DashboardController, :root_redirect)
     get("/dashboard", DashboardController, :index)
     get("/alerts/*id", AlertsController, :index)
-    get("/unauthorized", UnauthorizedController, :index)
   end
 
-  scope "/", ScreenplayWeb do
-    pipe_through([
-      :redirect_prod_http,
-      :browser,
-      :auth,
-      :ensure_auth,
-      :metadata,
-      :ensure_pa_message_admin
-    ])
-
-    get("/pa-messages", PaMessagesController, :index)
-    get("/pa-messages/new", PaMessagesController, :index)
-    get("/pa-messages/new/associate-alert", PaMessagesController, :index)
-    get("/pa-messages/:id/edit", PaMessagesController, :index)
-    get("/api/pa-messages/preview_audio", PaMessagesApiController, :preview_audio)
+  scope "/api", ScreenplayWeb do
+    pipe_through([:api, :authenticate])
+
+    get("/dashboard", DashboardApiController, :index)
+    get("/alerts", AlertsApiController, :index)
+    get("/alerts/non_access_alerts", AlertsApiController, :non_access_alerts)
   end
 
-  scope "/", ScreenplayWeb do
-    pipe_through([
-      :redirect_prod_http,
-      :browser,
-      :auth,
-      :ensure_auth,
-      :ensure_screens_admin,
-      :metadata
-    ])
+  # PA Message Management
 
-    get("/pending", ConfigController, :index)
-    get("/configure-screens/*app_id", ConfigController, :index)
+  scope "/pa-messages", ScreenplayWeb do
+    pipe_through([:browser, :authenticate])
+
+    get("/", PaMessagesController, :index)
+    get("/new", PaMessagesController, :index)
```

Comment:
Should the `/new` route now also use the `:ensure_pa_message_admin` plug?

Reply:
Since all of these route to the same action (`PaMessagesController.index`), I don't think it would make sense for one of them to be protected and the other not. The only thing these routes "do" is load the frontend app, which then takes over routing. On the backend, we only need to protect actions that will actually change data (assuming that _viewing_ things is always okay, which I think it is).

Reply:
Actually bringing this thread back (tangentially) - do you mind making an update to `NewPaMessage.tsx` on my behalf? Something this comment made me realize I overlooked was that an unauthenticated user could type in the url `https://screenplay.mbta.com/pa-messages/new` and be brought to an editable pa message form. Although they wouldn't be able to submit a new pa message now with this PR 🙏 , this still seems like an odd thing. We can just make another call to `isPaMessageAdmin()` and pass it to the PaMessageForm component.

Reply:
Not totally convinced it makes sense to support the "use case" of someone manually typing in URLs to get places they shouldn't... but I went ahead and did this and saw some other opportunities to clean up the role-checking code while I was at it. See the two new commits on this branch.